### PR TITLE
Modify Tube geometry to support exporting of the geometry 

### PR DESCRIPTION
**Pull request by Samsung-Blue**
1. Implemented Curve.getPoint function to support any tube geometry and also establish exporting of tube geometry which wasn't possible in the current version.
2. Exporting of tube geometry using ObjectLoader.js is now possible.
3. Example is included in the examples folder ( load_tubegeometry.html )

### CODE REVIEW

**Comment by mrdoob** [comment]
Sounds like you updated `ObjectLoader.js` but the file is not included in the PR. Do you mind doing some clean up? Also, please, do not include builds.

**Comment by Samsung-Blue** [comment]
Actually I have updated three.js file not ObjectLoader.js because tubeGeometry could not be exported as Curve.getPoint was not implemented in three.js

**Comment by Mugen87** [comment]
@Samsung-Blue Please read this article in order to improve your PR: [How to contribute to three.js](https://github.com/mrdoob/three.js/wiki/How-to-contribute-to-three.js)

File: docs/api/geometries/TubeBufferGeometry.html [hunk 1]
```diff
@@ -35,11 +35,12 @@
 		<h2>Example</h2>
 
 		<code>
-		function CustomSinCurve( scale ) {
-
-			THREE.Curve.call( this );
```
Comment (Mugen87) [inline docs/api/geometries/TubeBufferGeometry.html]: Why have you deleted this line? This breaks the inheritance of the curve object.

File: docs/api/geometries/TubeBufferGeometry.html [hunk 2]
```diff
@@ -48,9 +49,9 @@
 
 		CustomSinCurve.prototype.getPoint = function ( t ) {
 
-			var tx = t * 3 - 1.5;
-			var ty = Math.sin( 2 * Math.PI * t );
-			var tz = 0;
+			var tx = eval( this.xEq );
```
Comment (Mugen87) [inline docs/api/geometries/TubeBufferGeometry.html]: The usage of `eval` like this is bad practice...

File: docs/scenes/geometry-browser.html [hunk 3]
```diff
@@ -34,7 +34,7 @@
 
 		<a id='newWindow' href='./geometry-browser.html' target='_blank'>Open in New Window</a>
 
-		<script src="../../build/three.min.js"></script>
+		<script src="../../build/three.js"></script>
```
Comment (Mugen87) [inline docs/scenes/geometry-browser.html]: We don't want to use the unminified version of three.js here.

File: docs/scenes/js/geometry.js [hunk 4]
```diff
@@ -1302,46 +1303,6 @@ var guis = {
...
-		var folder = gui.addFolder( 'THREE.ExtrudeBufferGeometry' );
-
-		folder.add( data, 'steps', 1, 10 ).step( 1 ).onChange( generateGeometry );
-		folder.add( data, 'amount', 1, 20 ).step( 1 ).onChange( generateGeometry );
-		folder.add( data, 'bevelThickness', 1, 5 ).step( 1 ).onChange( generateGeometry );
-		folder.add( data, 'bevelSize', 1, 5 ).step( 1 ).onChange( generateGeometry );
-		folder.add( data, 'bevelSegments', 1, 5 ).step( 1 ).onChange( generateGeometry );
-
-		generateGeometry();
-
```
Comment (Mugen87) [inline docs/scenes/js/geometry.js]: Why do you have deleted this code?

File: examples/load_tubegeometry.html [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+				filename="./models/json/TubeGeometry.json";
+				var loader = new THREE.ObjectLoader();
+				loader.load(filename,function(obj){
+					 	scene.add(obj);
+				});
+				
+			}
+		</script>
+	</body>
+</html>
```
Comment (Mugen87) [inline examples/load_tubegeometry.html]: The entire code does not respect our code style...

**Review by Mugen87 — CHANGES_REQUESTED** [review]

**Comment by Mugen87** [comment]
I'm not sure this PR produces any useful benefits. Can you please explain your actual use case? `TubeGeometry` is already able to handle paths defined via curves..

**Pull request closed without merging**